helper.go

```diff
@@ -31,12 +30,30 @@ func decodeTimestamp(uuidBytes []byte) uint64 {
 
 // Helper function to parse and sanitize a UUID string.
 func parseUUID(uuid string) ([]byte, error) {
-	uuid = strings.ReplaceAll(uuid, "-", "")
-	if len(uuid) != 32 {
+	if len(uuid) == 32 {
+		// Fast path for UUIDs without dashes
+		return hex.DecodeString(uuid)
+	} else if len(uuid) == 36 {
+		// Validate dash positions
+		if uuid[8] != '-' || uuid[13] != '-' || uuid[18] != '-' || uuid[23] != '-' {
+			return nil, errors.New("invalid UUID format")
+		}
+	} else {
 		return nil, errors.New("invalid UUID length")
 	}
 
-	return hex.DecodeString(uuid)
+	// Remove dashes while copying characters
+	result := make([]byte, 32)
+	j := 0
+	for i := 0; i < len(uuid); i++ {
+		if uuid[i] == '-' {
+			continue
+		}
+		result[j] = uuid[i]
+		j++
+	}
```

Comment:
Why not a simple bytes.ReplaceAll?

Reply:
Good point! Primary reason was to address #10 . When I attempted 'Benchmarking', the current implementation was faster and avoids memory allocations. 

Here are the benchmark results comparing the two:

| **Implementation**      | **Time per Operation (ns/op)** | **Speed**   |
|--------------------------|---------------------------------|-------------      |
| `strings.ReplaceAll()`     | 146.2 ns/op                            | Baseline            |
| Current Approach        | 68.17 ns/op                            | **2.22x faster** |

**How?** 

This approach processes the `UUID` in one pass, directly copying characters into a pre-allocated byte slice. This avoids creating a new string, which is what `strings.ReplaceAll` does and keeps things lean.